reporters/cloud-run-scripts/CloudRunnerHub.js

```diff
@@ -69,34 +68,69 @@ app.post("/", async (req, res) => {
-      // promiseArray.push(_postJson(process.env.PROTOCOL_RUNNER_URL, botConfig));
-      promiseArray.push(_executeCloudRun(process.env.PROTOCOL_RUNNER_URL, botConfig));
+      const botConfig = _appendBlockNumberEnvVars(configObject[botName], lastQueriedBlockNumber, latestBlockNumber);
+      if (process.env.ENVIRONMENT == "production") {
+        promiseArray.push(_executeCloudRun(process.env.PROTOCOL_RUNNER_URL, botConfig));
+      } else {
+        promiseArray.push(_postJson(process.env.PROTOCOL_RUNNER_URL, botConfig));
+      }
     }
+    Logger.debug({
+      at: "CloudRunnerHub",
+      message: "Executing cloud run query from config file",
+      lastQueriedBlockNumber,
+      latestBlockNumber,
+      botsExecuted: Object.keys(configObject)
+    });
 
     // Loop through promise array and submit all in parallel. `allSettled` does not fail early if a promise is rejected.
+    // This `results` object will contain all information sent back from the spokes. This contains the process exit code,
+    // and importantly the full execution output which can be used in debugging.
     const results = await Promise.allSettled(promiseArray);
-    console.log(JSON.stringify(results)); // TODO: refine this logging with a winston log
+
+    Logger.debug({
+      at: "CloudRunnerHub",
+      message: "Batch execution promise resolved",
+      results
+    });
 
     // Validate that the promises returned correctly. If ANY have error, then catch them and throw them all together.
     let thrownErrors = [];
-    results.forEach(result => {
+    results.forEach((result, index) => {
       if (result.status == "rejected") {
-        thrownErrors.push(result);
+        // If the child process in the spoke crashed it will return 400 (rejected).
+        thrownErrors.push({
+          status: result.status,
+          execResponse: result.reason.response.data,
+          botIdentifier: Object.keys(configObject)[index]
+        });
+      } else if (result.value.execResponse.error) {
+        // If the child process exited correctly but contained an error.
+        thrownErrors.push({
+          status: result.status,
+          execResponse: result.value.execResponse,
+          botIdentifier: Object.keys(configObject)[index]
+        });
       }
     });
-
     if (thrownErrors.length > 0) {
       throw thrownErrors;
     }
 
     // If no errors and got to this point correctly then return a 200 success status.
-    res.status(200).send({ message: "All calls returned correctly", error: null });
-  } catch (error) {
-    console.log("error", error);
-    res.status(400).send({ message: "One or more calls failed with error", error: error });
+    Logger.debug({
+      at: "CloudRunnerHub",
+      message: "All calls returned correctly",
+      thrownErrors: null
+    });
+    res.status(200).send({ message: "All calls returned correctly", error: thrownErrors });
```

Comment:
Why are we passing thrownErrors on the success response? Shouldn't it always be empty?

Reply:
ye, it is an empty array if the response is a 200. I can change this to return `null` for the `error`.

Reply:
+1, yeah, I think that would be clearer.